### PR TITLE
Subscription auth where

### PR DESCRIPTION
**Pull request by angrykoala**
This PR adds support for `where` argument for subscriptions.
The behaviour will be the same as for where parameter (i.e. no nested where or operations)

`where` supports `jwt` params.

Behaviour between roles and where is still untested

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-graphql-docs-1475.surge.sh

File: docs/modules/ROOT/pages/auth/subscriptions.adoc [hunk 1]
```diff
@@ -23,7 +23,10 @@ Roles can be set for subscriptions. Only requests matching the roles set will be
 NOTE: Not Supported
 
 == Where
-NOTE: Not Supported
+If the rule `where` is set. The provided arguments will be used on xref::subscriptions/filtering.adoc[subscriptions filtering], regardless
```
Comment (mjfwebb) [inline docs/modules/ROOT/pages/auth/subscriptions.adoc:26]:
```suggestion
If the rule `where` is set the provided arguments will be used on xref::subscriptions/filtering.adoc[subscriptions filtering], regardless
```

File: docs/modules/ROOT/pages/subscriptions/filtering.adoc [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        title
+    }
+}
+```
+
+This way, only newly created movies with the genre `"Drama"` will trigger events to this subscription.
+
+
+=== Events
+Filtering behaves slightly differently depending on the event type:
```
Comment (tbwiss) [inline docs/modules/ROOT/pages/subscriptions/filtering.adoc:35]:
```suggestion
Filtering behaves slightly different depending on the event type:
```

File: packages/graphql/tests/e2e/subscriptions/auth/where.int.test.ts [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+                        name: "arthur",
+                        surname: "Conan Doyle",
+                    },
+                },
+            },
+        ]);
+        expect(wsClient.errors).toEqual([]);
+    });
+
+    test("where option in auth takes preference over where arg", async () => {
```
Comment (tbwiss) [inline packages/graphql/tests/e2e/subscriptions/auth/where.int.test.ts:164]: I assume this should be `precedence``
```suggestion
    test("where option in auth takes precedence over where arg", async () => {
```

**Review by tbwiss — DISMISSED** [review]
LGTM @angrykoala!

File: packages/graphql/src/schema/resolvers/subscriptions/subscribe.ts [hunk 4]
```diff
@@ -50,14 +52,28 @@ export function generateSubscribeMethod(node: Node, type: "create" | "update" |
...
+            const whereParams = { ...args.where, ...authWhere };
+            const matchWhereFilter = subscriptionWhere(whereParams, data[0]);
+            const matchUpdateDiffFilter = updateDiffFilter(data[0]);
+
+            return isSubscribedType && matchWhereFilter && matchUpdateDiffFilter;
         });
     };
 }
+
+function getAuthWhereParameters(authRules: AuthRule[], context: SubscriptionContext): Record<string, any> {
```
Comment (mjfwebb) [inline packages/graphql/src/schema/resolvers/subscriptions/subscribe.ts:73]: Is it possible to change the function return type so it doesn't use `any`?

File: packages/graphql/tests/e2e/subscriptions/auth/roles.int.test.ts [hunk 5]
```diff
@@ -18,7 +18,7 @@
  */
 
 import { Driver } from "neo4j-driver";
-import supertest, { Response } from "supertest";
+import { Response } from "supertest";
```
Comment (mjfwebb) [inline packages/graphql/tests/e2e/subscriptions/auth/roles.int.test.ts:21]:
```suggestion
import type { Response } from "supertest";
```

Can we specify it as a type import?

**Review by mjfwebb — COMMENTED** [review]
Just a couple of leading questions and they apply to all similar cases. No changes are requested, but I think we would benefit greatly from reducing the use of `any`.

**Comment by angrykoala** [comment]
Branch closed as it is stale

**Pull request closed without merging**